### PR TITLE
Support div prec incr

### LINKED ISSUE #8834: Support div_precision_increment attribute in DAGRequest
## Enhancement
https://github.com/pingcap/tidb/issues/51501

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #8834 

Problem Summary:

### Check List

Tests 

- [x] Unit test
Tried to add avg unit test, but failed. Will add integration test later.
- [ ] Integration test
Will add integration tests after tidb related pr merged.
- [ ] Manual test (add detailed scripts or steps below)
Currently, manually checked divide and avg with different div_precision_increment value locally.
43XVv8QeEr

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by SeaRise — APPROVED** [review]
lgtm

File: dbms/src/Functions/tests/gtest_div_precision_increment.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    context->getDAGContext()->setDivPrecisionIncrement(5);
+    /// Decimal32
+    ASSERT_COLUMN_EQ_V2(
+        createColumn<Nullable<Decimal64>>(std::make_tuple(13, 6), {"0.733333", "1.650000"}),
+        DB::tests::executeFunction(
+            *context,
+            func_name,
+            {createColumn<Decimal32>(std::make_tuple(8, 1), {"2.2", "3.3"}),
+             createColumn<Decimal32>(std::make_tuple(2, 0), {"3", "2"})}));
```
Comment (windtalker) [inline dbms/src/Functions/tests/gtest_div_precision_increment.cpp:60]: Add some tests to test the decimal infer result if DivPrecisionIncrement is set to some corner value like 0 and 30 ?

Comment (yibin87) [reply dbms/src/Functions/tests/gtest_div_precision_increment.cpp:60]: Added

**Review by windtalker — APPROVED** [review]
LGTM
